src/brain.coffee

```diff
@@ -398,35 +396,27 @@ class Brain
       else
         data.output = @master.errors.objectNotFound
 
-      # if we get a promise back and we are not in the async mode,
-      # leave an error message to suggest using an async version of rs
-      # otherwise, keep promises tucked into a list where we can check on
-      # them later
-      if async
-        promises.push
-          promise: if utils.isAPromise(data.output) then data.output else q(data.output)
-          data: data
-        continue
-      else if utils.isAPromise(data.output)
+      if not async and utils.isAPromise(data.output)
         data.output = "[ERR: Using async routine with reply: use replyAsync instead]"
```

Comment:
no - this is in a `for` loop, so you want to make sure to get the results of each callSignature before you do anything w/processing. `return` statements happen in lines 402 - 405 once all the calls have been made.